spec/rubocop/cop/style/if_with_boolean_literal_branches_spec.rb

```diff
@@ -0,0 +1,511 @@
+    end
+
+    it 'does not register an offense when using `unless condition` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless condition
+          true
+        else
+          false
+        end
+      RUBY
+    end
+  end
+
+  context 'when condition is a logical operator and operands do not known whether to return boolean value' do
+    it 'does not register an offense when using `if foo && bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo && bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo || bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo || bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo && bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo && bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo || bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo || bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo && bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo && bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo || bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo || bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo && bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo && bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo || bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo || bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+  end
+
+  context 'when complex condition' do
+    it 'registers and corrects an offense when using `if foo? && bar && baz?` with boolean literal branches' do
+      expect_offense(<<~RUBY)
+        if foo? && bar && baz?
+        ^^ Remove redundant `if` with boolean literal branches.
+          true
+        else
+          false
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        foo? && bar && baz?
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo? || bar || baz?` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo? || bar || baz?
```

Comment:
Why do you prefer not simplifying this one?

Reply:
This is because the return value changes from `true` to `"hi"` if `bar` is not a boolean value. Example:

```ruby
def foo?
  false
end

def bar
  "hi"
end

def baz?
  false
end

foo? || bar || baz? ? true : false #=> true
foo? || bar || baz? #=> "hi"
```

So this aims for safe detection, which keeps the return value as a boolean value.

Reply:
Ah, got it. Thanks @koic